syncd/FlexCounter.cpp

```diff
@@ -1396,6 +1513,16 @@ void FlexCounter::flexCounterThreadRunFunction()
     swss::RedisPipeline pipeline(&db);
     swss::Table countersTable(&pipeline, COUNTERS_TABLE, true);
 
+    string switchType;
+    swss::Table *appDbCountersTable = NULL;
+    getCfgSwitchType(switchType);
+    if (switchType == "voq")
+    {
+       swss::DBConnector appDb("CHASSIS_APP_DB", 0, true);
+       swss::RedisPipeline appDbPipeline(&appDb);
+       appDbCountersTable = new swss::Table(&appDbPipeline, "COUNTERS", true);
```

Comment:
don't use new operator !

Reply:
Why? I need a pointer and not a reference to an object here.

Reply:
use shared_ptr

Reply:
Done